pkg/util/helper/work_test.go

```diff
@@ -17,13 +17,272 @@ limitations under the License.
+					"metadata": map[string]interface{}{
+						"name": "test-deployment",
+					},
+				},
+			},
+			wantErr: true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			c := fake.NewClientBuilder().WithScheme(scheme)
+			if tt.existingWork != nil {
+				c = c.WithObjects(tt.existingWork)
+			}
+			client := c.Build()
+
+			err := CreateOrUpdateWork(context.TODO(), client, tt.workMeta, tt.resource)
+
+			if tt.wantErr {
+				assert.Error(t, err)
+				return
+			}
+			assert.NoError(t, err)
+			if tt.verify != nil {
+				tt.verify(t, client)
+			}
+		})
+	}
+}
+
+func TestGetWorksByLabelsSet(t *testing.T) {
+	scheme := runtime.NewScheme()
+	assert.NoError(t, workv1alpha1.Install(scheme))
+
+	work1 := &workv1alpha1.Work{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "work1",
+			Labels: map[string]string{
+				"app": "test",
+			},
+		},
+	}
+	work2 := &workv1alpha1.Work{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "work2",
+			Labels: map[string]string{
+				"app": "other",
+			},
+		},
+	}
+
+	tests := []struct {
+		name      string
+		works     []client.Object
+		labels    labels.Set
+		wantCount int
+	}{
+		{
+			name:      "no works exist",
+			works:     []client.Object{},
+			labels:    labels.Set{"app": "test"},
+			wantCount: 0,
+		},
+		{
+			name:      "find work by label",
+			works:     []client.Object{work1, work2},
+			labels:    labels.Set{"app": "test"},
+			wantCount: 1,
+		},
+		{
+			name:      "find multiple works",
+			works:     []client.Object{work1, work2},
+			labels:    labels.Set{},
+			wantCount: 2,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			client := fake.NewClientBuilder().
+				WithScheme(scheme).
+				WithObjects(tt.works...).
+				Build()
+
+			workList, err := GetWorksByLabelsSet(context.TODO(), client, tt.labels)
+			assert.NoError(t, err)
+			assert.Equal(t, tt.wantCount, len(workList.Items))
+		})
+	}
+}
+
+func TestGetWorksByBindingID(t *testing.T) {
+	scheme := runtime.NewScheme()
+	assert.NoError(t, workv1alpha1.Install(scheme))
+
+	bindingID := "test-binding-id"
+	work1 := &workv1alpha1.Work{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "work1",
+			Labels: map[string]string{
+				workv1alpha2.ResourceBindingPermanentIDLabel: bindingID,
+			},
+		},
+	}
+	work2 := &workv1alpha1.Work{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "work2",
+			Labels: map[string]string{
+				workv1alpha2.ClusterResourceBindingPermanentIDLabel: bindingID,
+			},
+		},
+	}
+
+	tests := []struct {
+		name       string
+		works      []client.Object
+		bindingID  string
+		namespaced bool
+		wantCount  int
+	}{
+		{
+			name:       "find namespaced binding works",
+			works:      []client.Object{work1, work2},
+			bindingID:  bindingID,
+			namespaced: true,
+			wantCount:  1,
+		},
+		{
+			name:       "find cluster binding works",
+			works:      []client.Object{work1, work2},
+			bindingID:  bindingID,
+			namespaced: false,
+			wantCount:  1,
```

Comment:
In both scenarios, the `wantCount` is 1, which doesn't distinguish between them. Is it possible to judge based on the content of the returned `workList`, such as `work.Name`?

Reply:
Yes, i'll add the judgements.